src/query/phrase_query/exact_phrase_scorer.rs

```diff
@@ -0,0 +1,291 @@
+use std::cmp::Ordering;
+
+use super::phrase_scorer::{PhraseScorer, PostingsWithOffset};
+use crate::docset::{DocSet, TERMINATED};
+use crate::fieldnorm::FieldNormReader;
+use crate::postings::Postings;
+use crate::query::bm25::Bm25Weight;
+use crate::query::{Intersection, Scorer};
+use crate::{DocId, Score};
+
+pub struct ExactPhraseScorer<TPostings: Postings> {
+    intersection_docset: Intersection<PostingsWithOffset<TPostings>, PostingsWithOffset<TPostings>>,
+    num_terms: usize,
+    left: Vec<u32>,
+    right: Vec<u32>,
+    phrase_count: u32,
+    fieldnorm_reader: FieldNormReader,
+    similarity_weight: Bm25Weight,
+    scoring_enabled: bool,
+    slop: u32,
+    used_slop: u32,
```

Comment:
slop specific attributes ;) This WIP commit would make us get rid of these in case of exact phrase scoring + get rid of the `fn has_slop()` .